src/types/bundle/index.ts

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { encoding as enc, pem } from '../../util';
+import { Entry, EntryKind } from '../rekor';
+import { Envelope } from './__generated__/envelope';
+import { Bundle } from './__generated__/sigstore_bundle';
+import {
+  HashAlgorithm,
+  VerificationMaterial,
+} from './__generated__/sigstore_common';
+import { TransparencyLogEntry } from './__generated__/sigstore_rekor';
+
+export * from './__generated__/envelope';
+export * from './__generated__/sigstore_bundle';
+export * from './__generated__/sigstore_common';
+
+const BUNDLE_MEDIA_TYPE =
+  'application/vnd.dev.sigstore.bundle+json;version=0.1';
+
+export const bundle = {
+  toDSSEBundle: (
+    envelope: Envelope,
+    certificate: string[],
+    rekorEntry: Entry
+  ): Bundle => ({
+    mediaType: BUNDLE_MEDIA_TYPE,
+    content: {
+      $case: 'dsseEnvelope',
+      dsseEnvelope: envelope,
```

Comment:
hm this disconnect from what's in the actual bundle is confusing.. took me a while to find this was defined in the autogenerated types from the protobuf. I wonder if this could be improved in the bundle?

Reply:
this might be a hangup from only working with json/json-schema directly in JS so this slight disconnect feels jarring on first encounter 😄

Reply:
Yeah, this was surprising to me as well when I first generated the types from the protobufs. The `$case` discriminator isn't present when the structure is serialized to JSON so it works out ok.

It actually does help when working with the data structure as you can just `switch` on the `$case` when trying to figure out which bundle "style" you're working with.